app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -78,8 +79,14 @@
     public static final int HISTORY_DELTA = 5000;
     public static final String FRAGMENT_TAG = "DocumentEditFragment";
     private static final String SAVESTATE_DOCUMENT = "DOCUMENT";
-    private static final String SAVESTATE_CURSOR_POS = "CURSOR_POS";
     private static final String SAVESTATE_PREVIEW_ON = "SAVESTATE_PREVIEW_ON";
+    public static DocDisplayType _docInitType;
+
+    private enum DocDisplayType {
+        FILE_MANAGER,
+        TODO,
+        QUICK_NOTE
+    }
```

Comment:
We don't need to track the filetype, as long we are at MainActivity it is OK to do the special handling. We don't care if it's a todo.txt, markdown etc

Reply:
> 
> 
> We don't need to track the filetype, as long we are at MainActivity it is OK to do the special handling. We don't care if it's a todo.txt, markdown etc. At same time, we also have filepath and appsettings for comparision. And for fragment visible detection there is also something.

I need to know what is used in the main activity (ToDo or QuickNote), because they both will be opened, and the keyboard will open for ToDo and QuickNote as a result keyboard will be bugged.
We need to open the keyboard only for active doc, and do not open for docs that in the background. That why I have created this detection.

Reply:
> and the keyboard will open for ToDo and QuickNote as a result keyboard will be bugged.

you can try without checking:
then try to open ToDo in the main activity then hide the app, and then back to the app.
At first, the keyboard will select the cursor position for ToDo and open the keyboard for ToDO, then for QuickNote (that in the background) as result will be a bug for ToDo will not show a keyboard after resume, and for QuickNote the keyboard will be shown as it opened last...

https://user-images.githubusercontent.com/56083544/124092942-0de8ac80-da60-11eb-80d1-c2ddf20d08ed.mp4

Reply:
OK I see, you need to know which of the files is opened currently. In this case you can also use AppSettings to get the filepath and compare it to the current Document.getFile